linker/shared/src/main/scala/org/scalajs/linker/frontend/optimizer/OptimizerCore.scala

```diff
@@ -5178,48 +5339,6 @@ private[optimizer] abstract class OptimizerCore(
     } else if (mutable) {
       withDedicatedVar(RefinedType(declaredType))
     } else {
-      def computeRefinedType(): RefinedType = bindingName match {
-        case _ if value.tpe.isExact || declaredType == AnyType =>
-          /* If the value's type is exact, or if the declared type is `AnyType`,
-           * the declared type cannot have any new information to give us, so
-           * we directly return `value.tpe`. This avoids a useless `isSubtype`
-           * call, which creates dependencies for incremental optimization.
-           *
-           * In addition, for the case `declaredType == AnyType` there is a
-           * stronger reason: we don't actually know that `this` is non-null in
-           * that case, since it could be the `this` value of a JavaScript
-           * function, which can accept `null`. (As of this writing, this is
-           * theoretical, because the only place where we use a declared type
-           * of `AnyType` is in `JSFunctionApply`, where the actual value for
-           * `this` is always `undefined`.)
-           */
-          value.tpe
-
-        case _: Binding.Local =>
-          /* When binding a something else than `this`, we do not receive the
-           * non-null information. Moreover, there is no situation where the
-           * declared type would bring any new information, since that would
-           * not be valid IR in the first place. Therefore, to avoid a useless
-           * call to `isSubtype`, we directly return `value.tpe`.
-           */
-          value.tpe
-
-        case Binding.This =>
-          /* When binding to `this`, if the declared type is not `AnyType`,
-           * we are in a situation where
-           * a) we know the value must be non-null, and
-           * b) the declaredType may bring more precise information than
-           *    value.tpe.base (typically when inlining a polymorphic method
-           *    that ends up having only one target in a subclass).
-           * We can refine the type here based on that knowledge.
-           */
-          val improvedBaseType =
-            if (isSubtype(value.tpe.base, declaredType)) value.tpe.base
-            else declaredType
-          val isExact = false // We catch the case value.tpe.isExact earlier
-          RefinedType(improvedBaseType, isExact, isNullable = false)
```

Comment:
Just to test my understanding: This is not necessary anymore, because we push down this knowledge onto local defs "from the beginning".

Reply:
Yes, that's right.